src/js/controller/instream-adapter.js

```diff
@@ -417,6 +418,10 @@ const InstreamAdapter = function(_controller, _model, _view, _mediaPool) {
         }
     }
 
+    function triggerAdBuffer(evt) {
+        _controller.trigger(AD_BUFFER, evt);
```

Comment:
Can we deal with this in `_handleStateChange`?

Reply:
Also see `_stateHandler` in ad-program-controller

```
    _stateHandler(state) {
        const { model } = this;
        switch (state) {
            case STATE_PLAYING:
            case STATE_PAUSED:
                model.set(PLAYER_STATE, state);
                break;
            default:
                break;
        }
    }
```

Reply:
and 

```
function normalizeState(newstate) {
    if (newstate === STATE_LOADING || newstate === STATE_STALLED) {
        return STATE_BUFFERING;
    }
    return newstate;
}
```

Reply:
What I expect to see is `_triggerAdPlayPause` renamed to `_triggerAdState` and add handling for buffering states
    function _triggerAdPlayPause(event) {
        const { newstate } = event;
        switch (newstate) {
            STATE_PLAYING:
            _controller.trigger(AD_PLAY, event);
           break;
          STATE_PAUSED:
            _controller.trigger(AD_PAUSE, event);
            break;
          STATE_LOADING:
          STATE_STALLED:
          STATE_BUFFERING:
            _controller.trigger(AD_BUFFER, event); // make any changes necessary to event properties
            break;
        }
    }